test/functional/ui/screen.lua

```diff
@@ -245,6 +249,21 @@ local ext_keys = {
 --              ignored, throw an error (use eq() or similar) to signal failure.
 -- any:         Lua pattern string expected to match a screen line.
 -- mode:        Expected mode as signaled by "mode_change" event
+-- unchanged:   Test that the screen state remain unchanged since the previous
+--              expect(...). Any flush event resulting in a different state is
+--              considered an error. Not observing any events until timeout
+--              is acceptable.
+-- intermediate:Test that the final state is the same as the previous expect,
+--              but expect an intermediate state that is different. If possible
+--              it is better to use an explicit screen:expect(...) for this
+--              intermediate state.
+-- redraw_bang: should be used for a test that follows immediately after a
+--              command("redraw!") or some other mechanism that will invoke
+--              redraw!. This will erase the screen internally, and check the
+--              entire screen got redraw from scratch.
```

Comment:
by "internally" you mean "internal to Screen.lua"? That confused me, because it's mentioned after discussion of `redraw!` being done in Nvim itself. Maybe this is clearer:

> redraw_bang: Clears the Screen state. Use this after Nvim's own internal screen was cleared (e.g. after `:redraw!`) to verify that the entire screen was redrawn.

Instead of `redraw_bang` how about `cleared` ?  While reading the tests,  `redraw_bang` looks like a command but it's actually a description of Nvim's state (IIUC).

Reply:
> No, I mean internally to the Screen in screen.lua, to prove that the state is in fact retransmitted, and not just left unchanged.

Regardless, it's not internal to `nvim`, which is what I was confused about.

> It is a description of the expected events in response to a redraw! command

Yes, so `redraw_bang` is confusing for the reasons I mentioned.

> cleared is a bit misleading, as a literal clear is not checked for,

Some other name then. `full_redraw` ?

Reply:
> Regardless, it's not internal to nvim, which is what I was confused about. Yes, so redraw_bang is confusing for the reasons I mentioned.

I just tried to explain what I meant, I never somehow denied the existing text could be confusing. 

> (Note that we don't need to mention that Screen.lua's state is cleared, that's an implementation detail I guess.)

Except that this precisely what this flag does. I will try to make the description more clear, but explaining what the flag does without mentioning what the flag does will not reduce confusion.

>  Or maybe just redrawn is unambiguous, in the context of Screen:expect().

Redraw just means redraw, which can be an incremental update,  and so is unambiguously worse than anything with "redraw!".